app/tests/test_browse.py

```diff
@@ -30,715 +26,1594 @@ def test_standard_user_redirected_to_browse_transferring_body_when_accessing_bro
+                <option value="first_body">{browse_files[0].consignment.series.body.Name}</option>
+                <option value="second_body">{browse_files[3].consignment.series.body.Name}</option>
+                <option value="third_body">{browse_files[10].consignment.series.body.Name}</option>
+                <option value="fourth_body">{browse_files[13].consignment.series.body.Name}</option>
+                <option value="fifth_body">{browse_files[19].consignment.series.body.Name}</option>
+                <option value="sixth_body">{browse_files[25].consignment.series.body.Name}</option>
+            </select>
+        """
+
+        assert_contains_html(
+            expected_html,
+            html,
+            "select",
+            {
+                "class": "govuk-select govuk-select__filters-form-transferring-body-select"
+            },
+        )
+
+    def test_browse_submit_search_query(
+        self, client: FlaskClient, mock_superuser, browse_files
+    ):
+        """
+        Given a superuser accessing the browse page
+        When they make a POST request
+        Then they should see results in content.
+        """
+        mock_superuser(client)
+
+        query = "test"
+        response = client.post("/browse", data={"query": query})
+
+        assert response.status_code == 200
+        assert b"Search for digital records" in response.data
+        assert b"Records found 6" in response.data
+
+    def test_browse_get_without_filter(
+        self, client: FlaskClient, mock_superuser, browse_files
+    ):
+        """
+        Given a superuser accessing the browse page
+        When they make a GET request with page as a query string parameter
+        Then they should see first five records on browse page content.
+        """
+        mock_superuser(client)
+
+        response = client.get("/browse")
+
+        assert response.status_code == 200
+        assert b"Search for digital records" in response.data
+        assert b"You are viewing" in response.data
+        assert b"Everything available to you" in response.data
+
+        soup = BeautifulSoup(response.data, "html.parser")
+        table = soup.find("table")
+        headers = table.find_all("th")
+        rows = table.find_all("td")
+
+        expected_results_table = [
+            [
+                "Transferring body",
+                "Series",
+                "Last record transferred",
+                "Records held",
+                "Consignments within series",
+            ],
+            [
+                "'fifth_body', 'fifth_series', '21/09/2023', '6', '2', "
+                "'first_body', 'first_series', '07/02/2023', '3', '2', "
+                "'fourth_body', 'fourth_series', '03/08/2023', '6', '2', "
+                "'second_body', 'second_series', '26/04/2023', '7', '2', "
+                "'sixth_body', 'sixth_series', '14/10/2023', '2', '1'"
+            ],
+        ]
 
+        assert [
```

Comment:
can we create an assertion function or 2 functions for the lines between 140 and 149 since we repeat these in many tests

Reply:
separated in a function to verify header and data rows and updated all test cases accordingly.